torch/_jit_internal.py

```diff
@@ -421,18 +448,43 @@ def fn(*args, **kwargs):
         else:
             return if_false(*args, **kwargs)
 
-    if if_true.__doc__ is None and if_false.__doc__ is not None:
-        doc = if_false.__doc__
-        if_true.__doc__ = doc
-    elif if_false.__doc__ is None and if_true.__doc__ is not None:
-        doc = if_true.__doc__
-        if_false.__doc__ = doc
-    elif if_false.__doc__ is None and if_true.__doc__ is None:
-        # neither function has a docstring
-        doc = None
+    if_true_signature = inspect.signature(if_true)
+    if_false_signature = inspect.signature(if_false)
+    if if_true_signature.parameters == if_false_signature.parameters:
+        parameters = list(if_true_signature.parameters.values())
     else:
-        raise RuntimeError("only one function can have a docstring")
-    fn.__doc__ = doc
+        if len(if_true_signature.parameters) != len(if_false_signature.parameters):
+            raise RuntimeError(
+                f"The number of parameters in signature of {getattr(if_true, '__name__', repr(if_true))} and "
+                f"{getattr(if_false, '__name__', repr(if_false))} mismatch: "
+                f"{len(if_true_signature.parameters)} != {len(if_false_signature.parameters)}."
+            )
+
+        parameters = [
+            inspect.Parameter(
+                name=extract_attribute(if_true_parameter, if_false_parameter, "name"),
+                kind=extract_attribute(if_true_parameter, if_false_parameter, "kind"),
+                default=extract_attribute(
+                    if_true_parameter, if_false_parameter, "default", sentinel=inspect.Parameter.empty
+                ),
+                annotation=extract_attribute(
+                    if_true_parameter, if_false_parameter, "annotation", sentinel=inspect.Parameter.empty
+                ),
+            )
+            for if_true_parameter, if_false_parameter in zip(
+                if_true_signature.parameters.values(), if_false_signature.parameters.values()
+            )
+        ]
+    return_annotation = extract_attribute(
+        if_true_signature,
+        if_false_signature,
+        "return_annotation",
+        sentinel=inspect.Signature.empty,
+        sentinel_if_unequal=True,
+    )
+    fn.__signature__ = inspect.Signature(parameters, return_annotation=return_annotation)  # type: ignore[attr-defined]
+
+    fn.__doc__ = extract_attribute(if_true, if_false, "__doc__")
```

Comment:
Just to clarify: this doesn't preserve the old behavior where it would set the docstring on `if_true` if `if_false` has a docstring, right?

Reply:
Correct. I've added the functionality back in the latest commits, so we don't need to worry about this.